### DISCUSSION TITLE
Consider setting `target' to es2022 in development to make inspecting private fields easier

### DISCUSSION THREAD

**Discussion by tomtiao** [vitejs/vite-plugin-react-swc#138] [Ideas]
When targeting es2020 in swc, private fields are transformed to `WeakMap`, and operations on them are converted to `_class_apply_descriptor_get` and `_class_apply_descriptor_set` functions, [as can be seen here](https://play.swc.rs/?version=1.3.74&code=H4sIAAAAAAAAA0vOSSwuVnBTqOblUgAC5bLEHF4uCDs5P6%2B4pKg0uSS%2FSEMTpgAESjIyi%2FVAKhVsFUyMIOK1MF1AYVTVRaklpUV5CE1w9bUAu2Ws1HwAAAA%3D&config=H4sIAAAAAAAAA1VPzQ6DIAy%2B8xSk5x2Mhx18hz0EYdVgQEiLyYjx3YcIbt76%2FfVrNyElzKxhkFseMwiKGOnCmeG0RPXJDKB2ijWZEOHR1JkPaVSWsVD7qUBUNGEsKe67vqsJsN4ztkTlnFnMmP47tXeBkPluPKxqmSzeG0VtBeffaxHrLzEFPC94ws%2FUyq7FYPjVkpFWFPsXkIDk%2FBcBAAA%3D), making it hard to inspect private fields in the debugger.

Is it possible to default `target` to es2022 or make `target` configurable?

![Transformed](https://github.com/vitejs/vite-plugin-react-swc/assets/47553271/e02b7d09-cbd2-4b9f-a4f9-9718e6347f39) ![Native](https://github.com/vitejs/vite-plugin-react-swc/assets/47553271/0a38cb57-4867-4af9-b6e0-0be77f699399)

**Comment by ArnaudBarre** [comment]
For now I've set this to something lower to allow people to use new feature while testing on Safari that be could not have implemented all features (less and less true actually)
I would like to avoid an option for this. Ideally we should have an option of "dev target" in Vite. WDYT @patak-dev @bluwy @sapphi-red ?

**Reply by bluwy** [reply]
Currently in Vite with esbuild, dev always processes jsx/ts files with `esnext`, unless the user change it via the `esbuild.target`. If we want to align the plugin with Vite core, we'd have to expose a target option in this plugin.

BUT there are callouts that this is confusing in https://github.com/vitejs/vite/issues/13756, so a dev target could make sense, but I'm not sure how it'll look like holistically if we also consider prebundling target, tsconfig target, etc.

**Comment by ArnaudBarre** [comment]
An option is available in the latest version!
